src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,470 @@
+    sleep 15
+    date
+    echo "Unstalling"
+    sudo echo '259:0 rbps=max wbps=max' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 20
+    date
+    echo "Stalling"
+    sudo echo '259:0 rbps=4 wbps=4' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 40
+    echo "UnStalling"
+    sudo echo '259:0 rbps=max wbps=max' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 600
+    exit
+done
+# eof
+~~~
+
+Next, make the script file executable:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+chmod 777 wal-flip.sh
+~~~
+
+Optionally, you can run a [workload]({% link {{ page.version.version }}/cockroach-workload.md %}) at this point, if you want to see your workload continue during WAL failover. However, the presence of the workload does not have any bearing on the operation of WAL failover.
+
+### 2. WAL Failover during transient disk stalls
+
+#### Cause transient disk stall
+
+Before triggering a WAL Failover, open the [DB Console]({% link {{ page.version.version }}/ui-overview.md %}) so that we can observe WAL Failover metrics.
+
+Next, trigger WAL Failover by introducing a transient disk stall that is shorter in duration than the value of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables). We will run the `wal-flip.sh` script that we created in [the previous step](#1-set-up-a-node-with-scripts-to-trigger-wal-failover).
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+wal-flip.sh
+~~~
+
+#### WAL Failover metrics
+
+{% include {{ page.version.version }}/wal-failover-metrics.md %}
+
+Whenever a WAL failover occurs on a disk, `wal.failover.switch.count` for the associated store will increment by 1.
+
+In [DB Console's Advanced Debug page]({% link {{ page.version.version }}/ui-custom-chart-debug-page.md %}), click on **Custom Time series chart**. On the custom chart page, add three charts: one for each of these metrics.
+
+Set the source of these metrics to be the node where you are running the disk stall/unstall script.
+
+<img src="{{ 'images/v24.3/wal-failover-metrics-chart.jpg' | relative_url }}" alt="WAL Failover Metrics Chart" style="border:1px solid #eee;max-width:100%" />
+
+Notice there is a switchover followed by each stall. The node with the stalled disk continues to perform normal operations, during and after WAL failover as the stalls are transient and shorter than the current value of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables).
+
+You can confirm that the node is still live and available using [`cockroach node status`]({% link {{ page.version.version }}/cockroach-node.md %}#node-status).
+
+### 3. WAL Failover during long disk stalls
+
+#### Cause long disk stall
+
+Before triggering a WAL Failover, open the [DB Console]({% link {{ page.version.version }}/ui-overview.md %}) so that we can observe WAL Failover metrics.
```

Comment:
```suggestion
Before triggering a WAL failover, open the [DB Console]({% link {{ page.version.version }}/ui-overview.md %}) so that you can observe WAL failover metrics.
```

Reply:
✅